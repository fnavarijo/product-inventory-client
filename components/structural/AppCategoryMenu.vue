<template>
  <div class="text-dark-gray-700">
    <div class="py-2 border-b border-gray-500 font-bold mb-2 text-lg">
      Productos
    </div>
    <ul>
      <li
        v-for="(category, index) in productsCategories"
        :key="index"
        class="pl-2 py-1 cursor-pointer"
        @click="select(category)"
      >
        <MenuOption
          :option="category"
          :sub-items="categories[category]"
        />
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

import MenuOption from '@/components/menu/MenuOption.vue';

import { ProductCategories } from '@/api/types';

export default Vue.extend({
  name: 'AppCategoryMenu',
  components: {
    MenuOption,
  },
  props: {
    categories: {
      type: Object as PropType<ProductCategories>,
      default: () => ({}),
    },
  },
  computed: {
    productsCategories (): Array<string> {
      return Object.keys(this.categories);
    },
  },
  methods: {
    select (option: string): void {
      this.$emit('select', option);
    },
  },
});
</script>
