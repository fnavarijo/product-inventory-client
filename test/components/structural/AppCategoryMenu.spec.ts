import { mount } from '@vue/test-utils';

import AppCategoryMenu from '@/components/structural/AppCategoryMenu.vue';

describe('AppCategoryMenu component', () => {
  it('should emit selected option on select', async () => {
    const wrapper = mount(AppCategoryMenu, {
      propsData: {
        categories: [
          'Accesories',
          'Clothing',
          'Shoes',
        ],
      },
    });

    const categoryElement = wrapper.find('li:nth-child(2)');
    await categoryElement.trigger('click');

    expect(wrapper.emitted('select')[0]).toEqual(['Clothing']);
  });

  it('should match snapshot', () => {
    const wrapper = mount(AppCategoryMenu, {
      propsData: {
        categories: [
          'Accesories',
          'Clothing',
          'Shoes',
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
