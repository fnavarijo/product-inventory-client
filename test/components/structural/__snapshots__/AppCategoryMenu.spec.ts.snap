// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCategoryMenu component should match snapshot 1`] = `
<div class="text-dark-gray-700">
  <div class="py-2 border-b border-gray-500 font-bold mb-2 text-lg">
    Productos
  </div>
  <ul>
    <li class="pl-2 py-1 cursor-pointer">
      <div class="flex flex-col min-h-6 justify-start"><span class="p-1 mb-2 option rounded-md capitalize select-none cursor-pointer hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
    0
  </span>
        <ul class="h-auto relative overflow-hidden transition-all duration-100 ease-in-slide flex-over-0">
          <transition-stub name="slide-down">
            <div class="relative top-0" style="display: none;">
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            A
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            c
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            c
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            e
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            s
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            o
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            r
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            i
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            e
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            s
          </span></li>
            </div>
          </transition-stub>
        </ul>
      </div>
    </li>
    <li class="pl-2 py-1 cursor-pointer">
      <div class="flex flex-col min-h-6 justify-start"><span class="p-1 mb-2 option rounded-md capitalize select-none cursor-pointer hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
    1
  </span>
        <ul class="h-auto relative overflow-hidden transition-all duration-100 ease-in-slide flex-over-0">
          <transition-stub name="slide-down">
            <div class="relative top-0" style="display: none;">
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            C
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            l
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            o
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            t
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            h
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            i
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            n
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            g
          </span></li>
            </div>
          </transition-stub>
        </ul>
      </div>
    </li>
    <li class="pl-2 py-1 cursor-pointer">
      <div class="flex flex-col min-h-6 justify-start"><span class="p-1 mb-2 option rounded-md capitalize select-none cursor-pointer hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
    2
  </span>
        <ul class="h-auto relative overflow-hidden transition-all duration-100 ease-in-slide flex-over-0">
          <transition-stub name="slide-down">
            <div class="relative top-0" style="display: none;">
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            S
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            h
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            o
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            e
          </span></li>
              <li class="subitems pl-2 py-1 capitalize border-l-2 border-dark-gray-700"><span class="py-1 px-2 rounded-md hover:bg-light-gray-600 hover:bg-opacity-50 transition duration-100 ease-linear">
            s
          </span></li>
            </div>
          </transition-stub>
        </ul>
      </div>
    </li>
  </ul>
</div>
`;
